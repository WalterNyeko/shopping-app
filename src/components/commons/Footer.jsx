import React from "react";
import { NavLink } from "react-router-dom";

const Footer = ({ departments }) => {
  return (
    <footer className="footer bg-dark sticky-bottom mb-auto py-3">
      <div className="container text-center h5 mt-4">
        {departments &&
          departments.length &&
          departments.map(({ name, department_id }) => (
            <NavLink
              to={`/department/${department_id}`}
              key={department_id}
              className="text-white mr-5"
            >
              {name}
            </NavLink>
          ))}
      </div>

      <div className="container text-center mt-4 mb-4">
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          <i className="fab fa-instagram fa-3x" />
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          <i className="fab fa-pinterest fa-3x" />
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          <i className="fab fa-twitter fa-3x" />
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          <i className="fab fa-facebook fa-3x" />
        </a>
      </div>

      <div className="container text-center mt-4 mb-4">
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          &copy; 2019 Shopmate Ltd
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          Contact
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          Privacy policy
        </a>
      </div>
    </footer>
  );
};
export default Footer;
