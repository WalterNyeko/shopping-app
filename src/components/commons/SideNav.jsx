import React, { Fragment } from 'react';
import '../../styles/SideNav.css';

 const SideNav = (props) => {
     const { content } = props;
    return (
        <Fragment>
            <div className="row">
                <div className="col-md-2">
                    <div className="sidenav">
                        <h6>Categories</h6>
                        <hr></hr>
                        <a href="#">Christmas</a>
                        <a href="#">Valentines</a>
                        <a href="#">Ladies</a>
                        <a href="#">Children</a>
                    </div>
                </div>
                <div className="col-md-10">
                    {content}
                </div>

            </div>
        </Fragment>
    )
}
export default SideNav;
