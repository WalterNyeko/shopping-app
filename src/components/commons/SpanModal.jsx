import React, { Fragment } from 'react';
import { Modal} from 'antd';



const ReusableModal = (props) => {
    const { 
        spanText, 
        modalTitle, 
        modalContent, 
        modalWidth,
        handleOk, 
        handleCancel,
        showModal,
        visible,
        confirmLoading,
        className,
    } = props;
    return (
      <Fragment>
        <span onClick={showModal} className={className}>
            {spanText}
        </span>
        <Modal
            title={modalTitle}
            visible={visible}
            onOk={handleOk}
            confirmLoading={confirmLoading}
            onCancel={handleCancel}
            width={modalWidth}
            footer={null}
        >
            <p>{modalContent}</p>
        </Modal>
      </Fragment>
    );
}
export default ReusableModal;