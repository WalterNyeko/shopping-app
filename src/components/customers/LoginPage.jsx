import React from 'react';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import useStyles from '../../styles/Signup';

const LoginPage = ({state: {email, password}, handleInputChange, handleSubmit }) => {
  const classes = useStyles();
  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
      <Typography component="h1" variant="h5">
          ShopMate
        </Typography>
        <form className={classes.form} noValidate>
        
          <Grid container spacing={2}>
            <Grid item xs={12}>
              <TextField
                variant="outlined"
                required
                fullWidth
                id="email"
                label="Email Address"
                name="email"
                autoComplete="email"
                defaultValue={email}
                onChange={handleInputChange}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                variant="outlined"
                required
                fullWidth
                name="password"
                label="Password"
                type="password"
                id="password"
                autoComplete="current-password"
                defaultValue={password}
                onChange={handleInputChange}
              />
            </Grid>
          </Grid>
          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
            onClick={handleSubmit}
          >
            Sign In
          </Button>
        </form>
      </div>
    </Container>
  );
}
export default LoginPage;