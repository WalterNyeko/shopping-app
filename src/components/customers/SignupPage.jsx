import React from 'react';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import useStyles from '../../styles/Signup';

const SignUp = ({ handleSubmit, handleInputChange, 
  state: {username, email, password}}) => {
  const classes = useStyles();
  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
      <Typography component="h1" variant="h5">
          ShopMate
        </Typography>
        <form className={classes.form} noValidate>
        
          <Grid container spacing={2}>
            <Grid item xs={12}>
              <TextField
                autoComplete="username"
                name="username"
                variant="outlined"
                required
                fullWidth
                id="username"
                label="Username"
                defaultValue={username}
                onChange={handleInputChange}
                autoFocus
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                variant="outlined"
                required
                fullWidth
                id="email"
                label="Email Address"
                defaultValue={email}
                onChange={handleInputChange}
                name="email"
                autoComplete="email"
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                variant="outlined"
                required
                fullWidth
                name="password"
                label="Password"
                defaultValue={password}
                onChange={handleInputChange}
                type="password"
                id="password"
                autoComplete="current-password"
              />
            </Grid>
          </Grid>
          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
            onClick={handleSubmit}
          >
            Sign Up
          </Button>
        </form>
      </div>
    </Container>
  );
}
export default SignUp;
