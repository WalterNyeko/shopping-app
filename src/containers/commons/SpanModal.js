import React, { Component, Fragment } from "react";
import SpanModalComponent from "../../components/commons/SpanModal";

export class SpanModal extends Component {
  state = {
    visible: false,
    confirmLoading: false
  };

  /**
   * triggers the modal to show up
   *
   * @returns {void}
   */
  showModal = () => {
    this.setState({
      visible: true
    });
  };

  /**
   * ensures that a modal state, visible is set to false
   * this therefore allows modal to hide
   *
   * @returns {void}
   */
  handleOk = () => {
    this.setState({
      visible: false
    });
  };

  /**
   * triggers the modal to hide
   * sets the state of the modal, visible to false
   *
   * @returns {void}
   */
  handleCancel = () => {
    this.setState({
      visible: false
    });
  };

  render() {
    const { visible, confirmLoading } = this.state;
    const {
      spanText,
      modalTitle,
      modalContent,
      modalWidth,
      className
    } = this.props;
    return (
      <Fragment>
        <SpanModalComponent
          visible={visible}
          confirmLoading={confirmLoading}
          spanText={spanText}
          modalTitle={modalTitle}
          modalContent={modalContent}
          modalWidth={modalWidth}
          showModal={this.showModal}
          handleCancel={this.handleCancel}
          handleOk={this.handleOk}
          className={className}
        />
      </Fragment>
    );
  }
}
