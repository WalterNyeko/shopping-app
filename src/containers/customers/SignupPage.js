import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import SignupPageComponent from "../../components/customers/SignupPage";
import { signupUser } from "../../store/actions/Customers";

export class SignupPage extends Component {
  state = {
    username: "",
    email: "",
    password: ""
  };

  /**
   * ensures the most recent value for each jsx input element is stored in the state
   * sets the state value of the input basing on its name attribute
   *
   * @param {Object} event
   *
   * @returns {void}
   */
  handleInputChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  /**
   * signs up a user into the app upon successful request
   *
   * @param {Object} event
   *
   * @returns {void}
   */
  handleSubmit = event => {
    event.preventDefault();
    const { username, email, password } = this.state;
    const userData = { name: username, email, password };
    const { signupUser } = this.props;
    signupUser(userData);
  };
  render() {
    return (
      <Fragment>
        <SignupPageComponent
          handleSubmit={this.handleSubmit}
          handleInputChange={this.handleInputChange}
          state={this.state}
        />
      </Fragment>
    );
  }
}
const mapStateToProps = state => ({
  username: state.username,
  email: state.email,
  password: state.password
});
export default connect(
  mapStateToProps,
  { signupUser }
)(SignupPage);
