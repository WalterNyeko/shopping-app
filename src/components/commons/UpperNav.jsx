import React, { Fragment } from "react";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import { NavLink } from "react-router-dom";
import bag from "../../images/shopping-bag.png";
import SearchIcon from "@material-ui/icons/Search";
import InputBase from "@material-ui/core/InputBase";
import { useStyles } from "../../styles/UpperNav";
import { SpanModal } from "../../../src/containers/commons/SpanModal";
import SignUpPage from "../../containers/customers/SignupPage";
import LoginPage from "../../containers/customers/LoginPage";

const UpperNav = ({ isLoggedIn, allDepartments, totalAmount }) => {
  const classes = useStyles();
  const { departments } = allDepartments;
  return (
    <Fragment>
      <nav className="navbar navbar-expand-md sticky-top navbar-light bg-light pt-0 pb-0 fir">
        <div className="navbar-brand">
          Hey!{" "}
          <a className="text-danger" href="#">
            <SpanModal
              spanText="Sign In"
              modalTitle="Login To Your ShopMate Account"
              modalWidth="450px"
              modalContent={<LoginPage />}
              className={classes.navItems}
            />
          </a>{" "}
          or{" "}
          <a href="#" className="text-danger">
            <SpanModal
              spanText="Register"
              modalTitle="Register For A ShopMate"
              modalWidth="450px"
              modalContent={<SignUpPage />}
              className={classes.navItems}
            />
          </a>
        </div>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarsExample04"
          aria-controls="navbarsExample05"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarsExample04">
          <ul className="navbar-nav m-auto nav-list">
            <li className="nav-item active">
              <a className="nav-link" href="#">
                Daily Deals <span className="sr-only">(current)</span>
              </a>
            </li>
            <li className="nav-item active">
              <a className="nav-link" href="#">
                Sell
              </a>
            </li>
            <li className="nav-item active">
              <a className="nav-link" href="#">
                Help & Contact
              </a>
            </li>
          </ul>
          <form className="form-inline my-2 my-md-0">
            <NavLink to="/mycart/" className="my-bag">
              <img src={bag} alt="bag" /> &nbsp; Your Bag &nbsp;{" "}
              <span className="text-danger"> $ {totalAmount}</span>
            </NavLink>
          </form>
        </div>
      </nav>
      <nav className="navbar navbar-expand-md fixed-top navbar-dark bg-dark nav-margin">
        <a className="navbar-brand logo" href="#">
          <NavLink to="/" className={classes.shopmateLogo}>
            SHOPMATE
          </NavLink>
        </a>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarsExample05"
          aria-controls="navbarsExample05"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarsExample05">
          <ul className="navbar-nav m-auto">
            {departments &&
              departments.map(({ department_id, name }) => (
                <Fragment key={department_id}>
                  <NavLink
                    to={`/department/${department_id}`}
                    className={classes.navItemsMainBar}
                  >
                    <li className="nav-item active">
                      <a className="nav-link nav-link-item" href="#">
                        {name}
                        <span className="sr-only">(current)</span>
                      </a>
                    </li>
                  </NavLink>
                  &nbsp;&nbsp;&nbsp;&nbsp;
                </Fragment>
              ))}
          </ul>
          <form className="form-inline my-2 my-md-0 text-white">
            <div className={classes.search}>
              <div className={classes.searchIcon}>
                <SearchIcon />
              </div>
              <InputBase
                placeholder="Searchâ€¦"
                classes={{
                  root: classes.inputRoot,
                  input: classes.inputInput
                }}
              />
            </div>
          </form>
        </div>
      </nav>
    </Fragment>
  );
};
export default UpperNav;
