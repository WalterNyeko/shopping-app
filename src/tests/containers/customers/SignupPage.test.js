import React from "react";
import { SignupPage } from "../../../containers/customers/SignupPage";
import { mount } from "enzyme";
import { SpanModal } from "../../../containers/commons/SpanModal";

const props = {
  signupUser: jest.fn()
};

describe("SignUp Page", () => {
  const wrapper = mount(<SignupPage {...props} />);
  const spanModalWrapper = mount(
    <SpanModal
      spanText="Sign Up"
      modalTitle="Sign UP For A ShopMate Account"
      modalWidth="450px"
      modalContent={<SignupPage {...props} />}
    />
  );
  it("should render signup page without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should show signup modal when user clicks Sign Up", () => {
    expect(spanModalWrapper.instance().state.visible).toBe(false);
    const signIn = spanModalWrapper.find("span");
    signIn.simulate("click"); //when user clicks the Sign Up text on nav bar (modal trigger)
    expect(spanModalWrapper.instance().state.visible).toBe(true);
  });

  it("should call signupUser() action method when user clicks Sign Up Button on the modal", () => {
    expect(spanModalWrapper.instance().state.visible).toBe(true); //modal is still open from previous click
    const spy = jest.spyOn(
      spanModalWrapper.instance().props.modalContent.props,
      "signupUser"
    );
    const signInButton = spanModalWrapper.find("span").at(8);
    signInButton.simulate("click"); //when user clicks the Sign Up Button on the modal
    expect(spy).toHaveBeenCalled();
  });
});
