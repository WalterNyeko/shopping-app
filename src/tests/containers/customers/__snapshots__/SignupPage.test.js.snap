// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Page should render signup page without crashing 1`] = `ReactWrapper {}`;
